CMakeLists.txt

```diff
@@ -159,10 +159,11 @@ endif()
 # Namespace
 
 if(NOT OCIO_NAMESPACE)
-	set(OCIO_NAMESPACE OpenColorIO CACHE STRING 
+	set(OCIO_NAMESPACE "OpenColorIO_v${OpenColorIO_VERSION_MAJOR}_${OpenColorIO_VERSION_MINOR}${OpenColorIO_VERSION_RELEASE_TYPE}" CACHE STRING 
```

Comment:
Based on experiences we have to be paranoiac so I think that a way to improve the namespace customization is not to override the name and the version parts. But it's:
```
if(NOT OCIO_NAMESPACE)
    set(OCIO_NAMESPACE "OpenColorIO_v${MAJOR}_${MINOR}${RELEASE_TYPE}" CACHE STRING  "Specific ...")
else()
    set(OCIO_NAMESPACE "OpenColorIO_&{OCIO_NAMESPACE}_v${MAJOR}_${MINOR}${RELEASE_TYPE}" CACHE STRING  "Specific ...")
endif()
```
For custom namespace the final namespace will be `OpenColorIO_CUSTOM_v2_0dev` for example. By doing so, the code provides a customization but still guaranties a minimal name clash. For example, an app. using OpenColorIO with a custom namespace but loading two different OpenColorIO shared libraries with conflicting versions i.e. v1 (in a plugin) vs. v2 (in the app itself).